CBLClient/Apps/CBLTestServer-Android/app/src/main/java/com/couchbase/CouchbaseLiteServ/server/RequestHandler/ReplicatorConfigurationRequestHandler.java

```diff
@@ -0,0 +1,260 @@
+package com.couchbase.CouchbaseLiteServ.server.RequestHandler;
+
+
+import android.util.Log;
+
+import com.couchbase.CouchbaseLiteServ.server.Args;
+import com.couchbase.CouchbaseLiteServ.server.Server;
+import com.couchbase.lite.Authenticator;
+// import com.couchbase.lite.ConflictResolver;
+import com.couchbase.lite.Database;
+import com.couchbase.lite.DatabaseEndpoint;
+import com.couchbase.lite.ReplicatorConfiguration;
+import com.couchbase.lite.URLEndpoint;
+import com.couchbase.litecore.C4Socket;
+
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Dictionary;
+import java.util.List;
+import java.util.Map;
+import java.io.InputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import javax.net.ssl.X509TrustManager;
+import javax.net.ssl.TrustManager;
+import java.security.cert.X509Certificate;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.HttpsURLConnection;
+import java.security.GeneralSecurityException;
+import android.content.res.AssetManager;
+import android.content.Context;
+
+
+public class ReplicatorConfigurationRequestHandler {
+
+    public ReplicatorConfiguration builderCreate(Args args) throws MalformedURLException, URISyntaxException {
+        Database sourceDb = args.get("sourceDb");
+        Database targetDb = args.get("targetDb");
+        URI targetURI = null;
+        if (args.get("targetURI") != null) {
+            targetURI = new URI((String) args.get("targetURI"));
+        }
+        if (targetDb != null){
+            DatabaseEndpoint target = new DatabaseEndpoint(targetDb);
+            return new ReplicatorConfiguration(sourceDb, target);
+        } else if (targetURI != null){
+            URLEndpoint target = new URLEndpoint(targetURI);
+            return new ReplicatorConfiguration(sourceDb, target);
+        } else {
+            throw new IllegalArgumentException("Incorrect configuration parameter provided");
+        }
+    }
+
+    public ReplicatorConfiguration configure(Args args) throws Exception {
+        Database sourceDb = args.get("source_db");
+        URI targetURL = null;
+        if (args.get("target_url") != null){
+            targetURL = new URI((String) args.get("target_url"));
+        }
+        Database targetDb = args.get("target_db");
+        String replicatorType = args.get("replication_type");
+        Boolean continuous = args.get("continuous");
+        List<String> channels = args.get("channels");
+        List<String> documentIds = args.get("documentIDs");
+        String pinnedservercert = args.get("pinnedservercert");
+        Authenticator authenticator = args.get("authenticator");
+        // ConflictResolver conflictResolver = args.get("conflictResolver");
```

Comment:
Can this commented code be removed?

Reply:
will wait for 6 months and remove, if dev get custom conflict resolver later , this might be needed